niworkflows/utils/bids.py

```diff
@@ -271,20 +272,10 @@ def collect_data(
                 # avoid clobbering layout.get
                 del layout_get_kwargs[entity]
 
-    if task:
-        queries['bold']['task'] = task
-
-    if echo:
-        queries['bold']['echo'] = echo
-
```

Comment:
rather than remove this, the `task` entity should also be added to the bids query